### PR TITLE
Tiny refactoring to reduce cloning

### PR DESCRIPTION
**Pull request by nazar-pc**
Not very meaningful, but it bothered me that `&Key` is only there to clone it later and in my code I have to throw away owned value that I could have provided directly.

It is more idiomatic to require owned value instead of reference that will be unconditionally cloned later.

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Reasoning makes sense to me!

Can you add a changelog entry please?

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 - Update to `libp2p-swarm` `v0.34.0`.
 
+- Require owned key in `get_record()` method (see [PR 2477])
```
Comment (mxinden) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Require owned key in `get_record()` method (see [PR 2477]).
```

**Review by mxinden — APPROVED** [review]
Looks good to me. Thanks @nazar-pc!

**Comment by mxinden** [comment]
Could you address the test failures @nazar-pc?

**Comment by nazar-pc** [comment]
Interestingly it didn't run all tests with `cargo test` when running at the root (probably because of the root package), should be fixed now.

**Comment by mxinden** [comment]
Could you also run Rust fmt :innocent:  ?

**Pull request merged**